transforms3d/quaternions.py

```diff
@@ -433,26 +428,30 @@ def quat2axangle(quat, identity_thresh=None):
     >>> np.allclose(theta, np.pi)
     True
 
-    If this is an identity rotation, we return a zero angle and an arbitrary
-    vector:
+    If this is an identity rotation, we return a zero angle and an
+    arbitrary vector
 
     >>> quat2axangle([1, 0, 0, 0])
     (array([ 1.,  0.,  0.]), 0.0)
 
     Notes
     -----
-    A quaternion for which x, y, z are all equal to 0, is an identity rotation.
-    In this case we return a 0 angle and an arbitrary vector, here [1, 0, 0]
+    A quaternion for which x, y, z are all equal to 0, is an identity
+    rotation.  In this case we return a 0 angle and an  arbitrary
+    vector, here [1, 0, 0]
     '''
     w, x, y, z = quat
     vec = np.asarray([x, y, z])
     if identity_thresh is None:
         try:
             identity_thresh = np.finfo(vec.dtype).eps * 3
-        except ValueError: # integer type
+        except ValueError:  # integer type
             identity_thresh = _FLOAT_EPS * 3
-    n = math.sqrt(x*x + y*y + z*z)
-    if n < identity_thresh:
+    len2 = x * x + y * y + z * z
+    if len2 < identity_thresh ** 2:
         # if vec is nearly 0,0,0, this is an identity rotation
         return np.array([1.0, 0, 0]), 0.0
-    return  vec / n, 2 * math.acos(w)
+    theta = 2 * math.acos(max(min(w, 1), -1))
```

Comment:
This constrains the input to acos to be between -1 and 1, and is a fix for quaternions that are not normalized.  Better fix here I believe:

https://github.com/matthew-brett/transforms3d/pull/5

Any chance of a review / comment on the issue on that pull request?